server/routes/bookings/bookingsController.ts

```diff
@@ -1,19 +1,24 @@
 import type { RequestHandler } from 'express'
-import { VisitService } from '../../services'
+import { BookerService, VisitService } from '../../services'
 import getPrisonInformation from '../../constants/prisonInformation'
 
 export default class BookingsController {
-  public constructor(private readonly visitService: VisitService) {}
+  public constructor(
+    private readonly visitService: VisitService,
+    private readonly bookerService: BookerService,
+  ) {}
 
   public view(): RequestHandler {
     return async (req, res) => {
       const { booker } = req.session
 
+      const prisoner = booker.prisoners ? booker.prisoners : await this.bookerService.getPrisoners(booker.reference)
```

Comment:
Might be good to cover this new dependency in `bookingsController.test.ts`. E.g.:
* in the current test, assert that `getPrisoners()` is **not** called
* duplicate the test but with `booker.prisoners` not set and assert that `getPrisoners()` is called as expected (this test could be minimal; i.e. don't need to re-check all the page elements)